damnit/backend/listener.py

```diff
@@ -117,24 +117,17 @@ def handle_event(self, record, msg: dict, run_data: RunData):
         self.db.ensure_run(proposal, run, record.timestamp / 1000)
         log.info(f"Added p%d r%d ({run_data.value} data) to database", proposal, run)
 
-        # Create subprocess to process the run
-        extract_proc = subprocess.Popen([
-            sys.executable, '-m', 'damnit.backend.extract_data',
-            str(proposal), str(run), run_data.value
-        ], cwd=self.context_dir, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-        self.extract_procs_queue.put((proposal, run, extract_proc))
+        log_path = process_log_path(run, proposal, self.context_dir)
+        log.info("Processing output will be written to %s",
+                 log_path.relative_to(self.context_dir))
 
-        # Create thread to log the subprocess's output
-        logger_thread = Thread(target=self.log_subprocess,
-                               args=(extract_proc.stdout, run, run_data))
-        logger_thread.start()
-
-    def log_subprocess(self, stdout_pipe, run, run_data):
-        with stdout_pipe:
-            for line_bytes in iter(stdout_pipe.readline, b""):
-                # Bytes to string, and remove trailing newline
-                line = line_bytes.decode().rstrip("\n")
-                log.info(f"r{run} ({run_data.value}): {line}")
+        with log_path.open('wb') as logf:
+            # Create subprocess to process the run
+            extract_proc = subprocess.Popen([
+                sys.executable, '-m', 'damnit.backend.extract_data',
+                str(proposal), str(run), run_data.value
+            ], cwd=self.context_dir, stdout=logf, stderr=subprocess.STDOUT)
```

Comment:
I had the same thought when I first wrote it. But when we launch the child process, it gets a copy of the file descriptor. Then the `with` block closes the file descriptor in the parent, but not the one in the child, so it can still write to the file. It's implicitly closed when the child exits.

Reply:
Ah hah, sneaky :octopus: